app/src/main/java/com/aaronjwood/portauthority/async/ScanHostsAsyncTask.java

```diff
@@ -40,17 +40,18 @@ public ScanHostsAsyncTask(MainAsyncResponse delegate) {
     protected Void doInBackground(String... params) {
         String ip = params[0];
         String parts[] = ip.split("\\.");
+        int hosts = Integer.decode(params[1]);
 
         ExecutorService executor = Executors.newCachedThreadPool();
 
         int SCAN_THREADS = 8;
-        int chunk = (int) Math.ceil((double) 255 / SCAN_THREADS);
+        int chunk = (int) Math.ceil((double) hosts / SCAN_THREADS);
         int previousStart = 1;
         int previousStop = chunk;
 
         for (int i = 0; i < SCAN_THREADS; i++) {
-            if (previousStop >= 255) {
-                previousStop = 255;
+            if (previousStop >= hosts) {
```

Comment:
You'll want to change `ScanHostsRunnable` as well, especially this https://github.com/aaronjwood/PortAuthority/blob/development/app/src/main/java/com/aaronjwood/portauthority/runnable/ScanHostsRunnable.java#L37

We'll need some logic somewhere to control how we loop through each octet based on the subnet.